submissions/morenko/task-frogger/js/app.js

```diff
@@ -0,0 +1,177 @@
+    minX;
+    maxY;
+    maxX;
+    
+    constructor( row, col ) {
+        
+        this.startX = (tileWidth * col) - (tileWidth);
+        this.startY = (tileHight * row) - (tileHight / 2);
+        this.x = this.startX;
+        this.y = this.startY;
+
+        this.maxY = tileHight * (countRows - 1);
+        this.maxX = tileWidth * (countColumns - 1);
+        this.minY = tileHight * (row - 1) - this.startY;
+        this.minX = 0;
+    }
+
+    reset() {
+        this.x = this.startX;
+        this.y = this.startY;
+        
+    }
+}
+
+class Enemy {
+
+    speed = (Math.floor(Math.random(4) * (500 - 100 + 1)) + 100);
+    sprite = 'images/Rock.png';
+
+    constructor ( PositionEnemy ) {
+        this.position = PositionEnemy;
+    }
+
+    update (dt) {
+        if (this.position.x <= this.position.maxX) {
+            this.position.run(this.speed, dt);
+        } else {
+            this.position.reset();
+        }
+    }
+
+    render = function() {
+        ctx.drawImage(Resources.get(this.sprite), this.position.x, this.position.y);
+    }
+}
+
+class PositionEnemy extends Position {
+
+    run(speed, dt) {
+        this.x += (speed * dt); 
+    }
+}
+
+class Player {
+    sprite = 'images/char-princess-girl.png';
+
+    constructor ( PositionPlayer ) {
+        this.position = PositionPlayer;
+    }
+
+    alertWin() {
+        if (this.position.y == this.position.minY) {
+            setTimeout(() => {
+                alert("Not bed! Try again!");
+                this.position.reset();
+            }, 0);
+        }
+    }
+
+    update () {
+        this.alertWin();
+    }
+
+    handleInput (keyCode) {
+
+        switch(keyCode) {
+            case 'up': 
+                this.position.moveUp();
+                break;
+            case 'down': 
+                this.position.moveDown();
+                break;
+            case 'right': 
+                this.position.moveRigtht();
+                break;
+            case 'left': 
+                this.position.moveLeft();
+                break;
+            default: 
+                break;
+        }
+    }
+
+    render () {
+        ctx.drawImage(Resources.get(this.sprite), this.position.x, this.position.y);
+    }
+}
+
+class PositionPlayer extends Position {
+
+    moveRigtht() {
+        let newPositionX = this.x + tileWidth;
+        if (newPositionX <= this.maxX) {
+            this.x = newPositionX; 
+        }
+    }
+
+    moveLeft() {
+        let newPositionX = this.x - tileWidth;
+        if (newPositionX >= this.minX) {
+            this.x = newPositionX; 
+        }
+    }
+
+    moveUp() {
+        let newPositionY = this.y - tileHight;
+        if (newPositionY >= this.minY) {
+            this.y = newPositionY; 
+        }
+    }
+
+    moveDown() {
+        let newPositionY = this.y + tileHight;
+        if (newPositionY <= this.maxY) {
+            this.y = newPositionY;
+        }
+    }
+}
+
+const player = new Player (new PositionPlayer( 5, 3 ));
```

Comment:
From the perspective of your Player's class user, it seems strange to ask for a `PositionPlayer` object as an argument. Wouldn't it be better, if `Player` class would just receive coordinates as arguments, and `PositionPlayer` would be created in `Player`'s constructor instead?